ocs_ci/deployment/deployment.py

```diff
@@ -1198,3 +1263,308 @@ def setup_persistent_monitoring():
+                )
+            index = +1
+        # Check for RBD mirroring pods
+        for cluster in get_non_acm_cluster_config():
+            config.switch_ctx(cluster.MULTICLUSTER["multicluster_index"])
+            mirror_pod = get_pod_count(label="app=rook-ceph-rbd-mirror")
+            if not mirror_pod:
+                raise PodNotCreated(
+                    f"RBD mirror pod not found on cluster: "
+                    f"{cluster.ENV_DATA['cluster_name']}"
+                )
+            self.enable_csi_sidecar()
+
+        # Reset CTX back to ACM
+        config.switch_acm_ctx()
+
+    def configure_volume_replication_class(self):
+        """
+        Configure volume replication class on all non-acm clusters
+        beginning ctx: ACM
+
+        """
+        for cluster in get_non_acm_cluster_config():
+            config.switch_ctx(cluster.MULTICLUSTER["multicluster_index"])
+            volume_replication_data = templating.load_yaml(
+                constants.VOLUME_REPLICATION_CLASS
+            )
+            volume_replication_yaml = tempfile.NamedTemporaryFile(
+                mode="w+", prefix="rbd_volumereplication_", delete=False
+            )
+            templating.dump_to_temp_yaml(
+                volume_replication_data,
+                volume_replication_yaml.name,
+            )
+            run_cmd(f"oc create -f {volume_replication_yaml.name}")
+        config.switch_acm_ctx()
+
+    def enable_csi_sidecar(self):
+        """
+        Enable sidecar containers for rbd mirroring on each of the
+        ODF cluster
+
+        """
+        patch_cmd = (
+            f"oc patch cm rook-ceph-operator-config "
+            f"-n {constants.OPENSHIFT_STORAGE_NAMESPACE}"
+            f"--type json --patch {constants.RBD_SIDECAR_PATCH_CMD}"
+        )
+        run_cmd(patch_cmd)
+        # Number of containers should be 8/8 from 2 pods now which makes total 16 containers
+        rbd_pods = (
+            f"oc get pods -n {constants.OPENSHIFT_STORAGE_NAMESPACE} "
+            f"-l app=csi-rbdplugin-provisioner -o jsonpath={{.items[*].spec.containers[*].name}}"
+        )
+        out = run_cmd(rbd_pods)
+        if constants.rbd_sidecar_count != len(out.split(" ")):
+            raise RBDSideCarContainerException("")
+
+    def validate_mirror_peer(self):
+        """
+        Validate mirror peer,
+        Begins with CTX: ACM
+
+        1. Check initial phase of 'ExchangingSecret'
+        2. Check token-exchange-agent pod in 'Running' phase
+
+        Raises:
+            ResourceWrongStatusException: If pod is not in expected state
+
+        """
+        # Check mirror peer status only on HUB
+        mirror_peer = ocp.OCP(
+            kind="MirrorPeer", namespace=constants.DR_DEFAULT_NAMESPACE
+        )
+        # Need to satisfy both the phases
+        mirror_peer.wait_for_phase(phase="ExchangingSecret")
+        mirror_peer.wait_for_phase(phase="ExchangedSecret")
```

Comment:
IMO we should not wait for both in serial order here.
Either one of `ExchangedSecret` and `ExchangingSecret` is an acceptable result.

Reply:
done, now we are waiting for either one of them